File: java/FindDuplicates.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.HashMap;
import java.util.Map;

public class FindDuplicates {

    public static List<Integer> findModeNestedLoops(List<Integer> l) {
        List<Integer> duplicates = new ArrayList<Integer>();
        for(int i = 0; i < l.size(); i++){
            for(int j = i + 1; j < l.size(); j++){
                if(l.get(i) == l.get(j)){
                    duplicates.add(l.get(i));
                    break;
                }
            }
        }
        return duplicates;
    }

    public static List<Integer> findModeMap(List<Integer> l){
        List<Integer> duplicates = new ArrayList<Integer>();
        Map<Integer, Integer> aux = new HashMap<Integer, Integer>();
        Integer temp;
        for(Integer i : l){
            temp = aux.get(i);
            if(temp == null){
                aux.put(i, i);
            }else{
                duplicates.add(temp);
            }
        }
        return duplicates;
    }

    public static void main(String[] args) {
        // some test strings:
        List<Integer> sample1 = new ArrayList<Integer>(Arrays.asList(3, 7, 5, 6, 7, 4, 8, 5, 7, 66));
        List<Integer> sample2 = new ArrayList<Integer>(Arrays.asList(3, 5, 6, 4, 4, 5, 66, 6, 7, 6));
        List<Integer> sample3 = new ArrayList<Integer>(Arrays.asList(3, 0, 5, 1, 0));
        List<Integer> sample4 = new ArrayList<Integer>(Arrays.asList(3));
        System.out.println("Sample 1: " + findModeNestedLoops(sample1));
        System.out.println("Sample 2: " + findModeNestedLoops(sample2));
        System.out.println("Sample 3: " + findModeNestedLoops(sample3));
        System.out.println("Sample 4: " + findModeNestedLoops(sample4));
        System.out.println("Sample 1: " + findModeMap(sample1));
        System.out.println("Sample 2: " + findModeMap(sample2));
        System.out.println("Sample 3: " + findModeMap(sample3));
        System.out.println("Sample 4: " + findModeMap(sample4));
    }

}

File: python/find_duplicates.py
def find_duplicates_nested_loop(l: list) -> list:
    duplicates = []
    for i in range(len(l)):
        for j in range(len(l) - i - 1):
            if l[i] == l[i + j + 1]:
                duplicates.append(l[i])
                break
    return duplicates

def find_duplicates_map(l: list) -> list:
    duplicates = []
    aux = set()
    for i in l:
        if i in aux:
            duplicates.append(i)
        else:
            aux.add(i)
    return duplicates

if __name__ == "__main__":
    sample1 = [3, 7, 5, 6, 7, 4, 8, 5, 7, 66]
    sample2 = [3, 5, 6, 4, 4, 5, 66, 6, 7, 6]
    sample3 = [3, 0, 5, 1, 0]
    sample4 = [3]
    
    print("Sample 1:", find_duplicates_nested_loop(sample1))
    print("Sample 2:", find_duplicates_nested_loop(sample2))
    print("Sample 3:", find_duplicates_nested_loop(sample3))
    print("Sample 4:", find_duplicates_nested_loop(sample4))

    print("Sample 1:", find_duplicates_map(sample1))
    print("Sample 2:", find_duplicates_map(sample2))
    print("Sample 3:", find_duplicates_map(sample3))
    print("Sample 4:", find_duplicates_map(sample4))